### ISSUE TITLE
Subscriptions with optional parameters only work as a filters on specific values of those params

### ISSUE THREAD

**Issue by rduhard** [aws-amplify/amplify-cli#3418]
**Describe the bug**
Codegen for iOS (swift) doesn't allow for subscriptions with optional parameters to be omitted, only for them to be nil. According to this: https://docs.aws.amazon.com/appsync/latest/devguide/real-time-data.html#argument-null-value-has-meaning, sending "null" vs not sending the param at all is treated differently. One filters on a null value, the other does not filter at all. Currently we are unable to achieve the latter because it is always sent as a nil value, and we're unable to omit the filter param altogether with the given API.

**Amplify CLI Version**
4.13.2

**To Reproduce**

1. Create a subscription that takes in 1 or more optional parameters
2. run codegen to generate API.swift
3. See that only one version of the subscription is generated that takes in optional params, which default to nil if not provided
4. When subscribing via the `subscribe` method, the `variables` object is passed to the server as `[**param1**: _null_]`

**Expected behavior**
Multiple subscription models that would allow for passing in both filter on nil and no filter at all. We would like to both filter on specific values (something or nil) and also to not filter i.e. 

1. `give me all messages with id="1"`
2. `give me all messages with id=null`
3. `give me all messages`

Currently we can only accomplish 1. and 2. but not 3. with the generated subscription model
My assumption is that for this to work, we'd need to generate more than one subscription model ( # of params ^ 2 to be exact) as opposed to a single one that takes in optional values. This would allow for the all 3 cases above to both be feasible. 

**Screenshots**
Schema definition
```
onDeleteMessage(messageThreadId: ID, messageId: ID): Message
	@aws_subscribe(mutations: ["deleteMessage"])
```

![image](https://user-images.githubusercontent.com/5777653/74383771-d5d6c000-4dbd-11ea-904f-54acf76579e9.png)
This is the code generated for a subscription that takes in 2 optional parameters. The problem is the `variables` property is always defined for both params, so there is no option provided to not filter on one or both of those params, only to filter on those parameters being a specific value (including the value nil).

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version. v13.3.0

**Additional context**

**Comment by edwardfoyle** [comment]
Hi @rduhard does adding another subscription on the same operation but with no input arguments solve your use case? Something like `allMessagesOnDelete: Message @aws_subscribe(mutations: ["deleteMessage"])`

**Comment by rduhard** [comment]
@edwardfoyle Yes that's exactly the way we've solved it for now. So in order to achieve the ability to not filter on a param and also filter on a value, we have to have n^2 subscriptions defined (where n is the number of optional/filterable parameters.) So it's not our ideal solution as it really bloats the schema. Was thinking that the codegen could automatically create those options without having to have them explicit in the schema since I think we should be able to both filter on nil/value and not filter at all with an optional param.

**Comment by edwardfoyle** [comment]
@rduhard If you really need every combination of inputs to be optional, then that is how you would have to do it. But if only certain inputs are optional, that should significantly reduce the number of subscriptions you would have to create. We will add this issue as an enhancement.

**Comment by IsaacTrevino** [comment]
I don't see why parameters are generated automatically, for the generic onCreate/onUpdate/onDelete, I create my own subscriptions for which I need specific requirements.

**Inside my amplify schema.graphql build folder**

Amplify Appsync Generated by schema which forced me to delete the parameter `employeeID` when I use it in my client, Otherwise it won't return subscribed data.
```graphql
  onCreateTodo(employeeID: String): Todo @aws_subscribe(mutations: ["createTodo"])
  onUpdateTodo(employeeID: String): Todo @aws_subscribe(mutations: ["updateTodo"])
  onDeleteTodo(employeeID: String): Todo @aws_subscribe(mutations: ["deleteTodo"])
```
My custom Graphql Schema creates this which is perfect, and the only one I need for precise subscribing.
```graphql
  subCreateTodo(employeeID: ID!): Todo @aws_subscribe(mutations: ["createTodo"])
  subUpdateTodo(employeeID: ID!): Todo @aws_subscribe(mutations: ["updateTodo"])
  subDeleteTodo(employeeID: ID!): Todo @aws_subscribe(mutations: ["deleteTodo"])
```

**Comment by Sedriq** [comment]
Faced the same problem. I suppose choosing what combinations of arguments should be used during code generation for particular subscription would be a useful feature
